src/test/java/org/alfresco/repo/site/SiteServiceImplTest.java

```diff
@@ -1586,40 +1598,67 @@ public Object execute() throws Throwable
+
+                authenticationComponent.setCurrentUser(USER_ONE);
+                PagingResults<SiteMembership> siteM = siteService.listSitesPaged(USER_ONE, null, new PagingRequest(1000));
+                assertNotNull(siteM);
+                assertFalse(siteM.hasMoreItems());
 
                 // Create a site as user two and add user one
                 authenticationComponent.setCurrentUser(USER_TWO);
                 siteService.createSite(TEST_SITE_PRESET, siteName2, TEST_TITLE, TEST_DESCRIPTION, SiteVisibility.PRIVATE);
-                siteService.setMembership("testMembership2", USER_ONE, SiteModel.SITE_CONSUMER);
+                siteService.setMembership(siteName2, USER_ONE, SiteModel.SITE_CONSUMER);
 
                 // Create a site as user three and add user one
                 authenticationComponent.setCurrentUser(USER_THREE);
                 siteService.createSite(TEST_SITE_PRESET, siteName3, TEST_TITLE, TEST_DESCRIPTION, SiteVisibility.PRIVATE);
-                siteService.setMembership("testMembership3", USER_ONE, SiteModel.SITE_COLLABORATOR);
+                siteService.setMembership(siteName3, USER_ONE, SiteModel.SITE_COLLABORATOR);
 
                 authenticationComponent.setCurrentUser(USER_ONE);
                 members = siteService.listSiteMemberships(USER_ONE, 0);
                 assertNotNull(members);
-                assertEquals(3, members.size());
+                assertEquals(5, members.size());
                 assertEquals(USER_ONE, members.get(0).getPersonId());
                 assertEquals(SiteModel.SITE_MANAGER, members.get(0).getRole());
                 assertEquals(siteName1, members.get(0).getSiteInfo().getShortName());
                 assertEquals(USER_ONE, members.get(1).getPersonId());
                 assertEquals(SiteModel.SITE_CONSUMER, members.get(1).getRole());
-                assertEquals(siteName2, members.get(1).getSiteInfo().getShortName());
+                assertEquals(publicSite, members.get(1).getSiteInfo().getShortName());
                 assertEquals(USER_ONE, members.get(2).getPersonId());
-                assertEquals(SiteModel.SITE_COLLABORATOR, members.get(2).getRole());
-                assertEquals(siteName3, members.get(2).getSiteInfo().getShortName());
+                assertEquals(SiteModel.SITE_CONTRIBUTOR, members.get(2).getRole());
+                assertEquals(moderatedSite, members.get(2).getSiteInfo().getShortName());
+                assertEquals(USER_ONE, members.get(3).getPersonId());
+                assertEquals(SiteModel.SITE_CONSUMER, members.get(3).getRole());
+                assertEquals(siteName2, members.get(3).getSiteInfo().getShortName());
+                assertEquals(USER_ONE, members.get(4).getPersonId());
+                assertEquals(SiteModel.SITE_COLLABORATOR, members.get(4).getRole());
+                assertEquals(siteName3, members.get(4).getSiteInfo().getShortName());
 

```

Comment:
Start of the actual test of listing site memberships authenticated as different users.